roachpb/data.go

```diff
@@ -112,11 +112,15 @@ func MakeKey(keys ...[]byte) []byte {
 // BytesNext returns the next possible byte by appending an \x00.
 func BytesNext(b []byte) []byte {
 	// TODO(spencer): Do we need to enforce KeyMaxLength here?
-	return append(append([]byte(nil), b...), 0)
+	bn := make([]byte, len(b)+1)
```

Comment:
You might want to add a comment referring to the PR so that this isn't accidentally reverted.